### PR TITLE
 [Prow CI workflow] Add hypershift-agent-manual chain and job for conformance test

### PR DESCRIPTION
**Pull request by LiangquanLi930**
**Why to add this chain:**
+ We had 2 options to add Agent based on [hypershift guidance](https://hypershift-docs.netlify.app/how-to/agent/create-agent-cluster/#adding-agents).
+ Currently, we had lots of bugs pending on Metal3 issue, cannot verify them.

**What I did in this PR:**
+ Add the CI to add agent through manual
+ Adjust the directory structure
+ Add manual's CI job

### CODE REVIEW

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-mce-agent-connected-ipv4-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-mce-agent-connected-ipv4-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-mce-agent-connected-ipv4-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-mce-agent-connected-ipv4-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-mce-agent-connected-dualstack-manual-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-mce-agent-connected-ipv4-manual-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-mce-agent-connected-dualstack-manual-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-mce-agent-connected-ipv4-manual-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-mce-agent-connected-dualstack-manual-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-mce-agent-connected-ipv4-manual-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-mce-agent-connected-ipv4-manual-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-mce-agent-connected-ipv4-manual-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-mce-agent-connected-ipv4-manual-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-mce-agent-connected-ipv4-manual-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-mce-agent-connected-ipv4-manual-conformance

File: ci-operator/step-registry/hypershift/mce/agent/metal3/conformance/hypershift-mce-agent-metal3-conformance-workflow.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  steps:
+    allow_best_effort_post_steps: true
+    allow_skip_on_success: true
+    post:
+    - chain: hypershift-mce-agent-destroy
+    - chain: baremetalds-ofcir-post
+    test:
+    - chain: hypershift-conformance
+    pre:
+    - chain: baremetalds-ofcir-pre
```
Comment (jparrill) [inline ci-operator/step-registry/hypershift/mce/agent/metal3/conformance/hypershift-mce-agent-metal3-conformance-workflow.yaml:16]: This is the typo I mentioned in the overall review comment

File: ci-operator/step-registry/hypershift/mce/agent/metal3/conformance/hypershift-mce-agent-metal3-conformance-workflow.yaml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  documentation: |-
+    The HyperShift mce agent conformance (connected) workflow executes tests against a new ephemeral
+    The installation method for nodepool is metal3.
+    The current workflow supports both IPv4 and dual-stack
+  steps:
+    allow_best_effort_post_steps: true
+    allow_skip_on_success: true
+    post:
+    - chain: hypershift-mce-agent-destroy
+    - chain: baremetalds-ofcir-post
```
Comment (jparrill) [inline ci-operator/step-registry/hypershift/mce/agent/metal3/conformance/hypershift-mce-agent-metal3-conformance-workflow.yaml:12]: This is the typo I mentioned in the overall review comment

File: ci-operator/step-registry/hypershift/agent/create/add-worker-metal3/hypershift-agent-create-add-worker-metal3-ref.metadata.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
+{
+	"path": "hypershift/agent/create/add-worker-metal3/hypershift-agent-create-add-worker-metal3-ref.yaml",
+	"owners": {
+		"approvers": [
```
Comment (jparrill) [inline ci-operator/step-registry/hypershift/agent/create/add-worker-metal3/hypershift-agent-create-add-worker-metal3-ref.metadata.json:4]: Missing @bryan-cox and myself (@jparrill) in the list :/

File: ci-operator/step-registry/hypershift/agent/create/add-worker-metal3/OWNERS [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+- csrwng
+- enxebre
+- sjenning
+- imain
+- LiangquanLi930
+- heliubj18
+- eranco74
+- avishayt
+options: {}
+reviewers:
```
Comment (jparrill) [inline ci-operator/step-registry/hypershift/agent/create/add-worker-metal3/OWNERS:11]: Missing @bryan-cox and myself (@jparrill) in the list :/

File: ci-operator/step-registry/hypershift/agent/create/add-worker-metal3/OWNERS [hunk 5]
```diff
@@ -0,0 +1,19 @@
+approvers:
```
Comment (jparrill) [inline ci-operator/step-registry/hypershift/agent/create/add-worker-metal3/OWNERS:1]: Missing @bryan-cox and myself (@jparrill) in the list :/

File: ci-operator/step-registry/cucushift/installer/rehearse/baremetalds/assisted/hypershift-agent/mce/provision/cucushift-installer-rehearse-baremetalds-assisted-hypershift-agent-mce-provision-chain.yaml
Comment (jparrill) [inline ci-operator/step-registry/cucushift/installer/rehearse/baremetalds/assisted/hypershift-agent/mce/provision/cucushift-installer-rehearse-baremetalds-assisted-hypershift-agent-mce-provision-chain.yaml:1]: The filename contains a typo `baremetalds`

**Review by jparrill — CHANGES_REQUESTED** [review]
The only concern I have is that this job modification, looks like it will overwrite the standard workflow of the Agent and that could not happen. This is a temporary solution to overcome the issue we've dealing with in Metal3. So I want to make sure that this is just a temporary solution and we will come back at some point to the BMH and Metal3 workflow.

As I mentioned in Slack, I could agree if we cover both and all of us agree with that, but I would say that is not supported at all. WDYT @celebdor ?

small nit: The file name contains a typo: `baremetalds` also in the `agent-metal3-conformance-workflow` file.

File: ci-operator/step-registry/hypershift/mce/agent/metal3/conformance/hypershift-mce-agent-metal3-conformance-workflow.yaml (hunk 1 above)
Comment (LiangquanLi930) [reply ci-operator/step-registry/hypershift/mce/agent/metal3/conformance/hypershift-mce-agent-metal3-conformance-workflow.yaml:16]: baremetalds is as expected, where 'ds' stands for https://github.com/openshift-metal3/dev-scripts

File: ci-operator/step-registry/cucushift/installer/rehearse/baremetalds/assisted/hypershift-agent/mce/provision/cucushift-installer-rehearse-baremetalds-assisted-hypershift-agent-mce-provision-chain.yaml
Comment (LiangquanLi930) [reply ci-operator/step-registry/cucushift/installer/rehearse/baremetalds/assisted/hypershift-agent/mce/provision/cucushift-installer-rehearse-baremetalds-assisted-hypershift-agent-mce-provision-chain.yaml:1]: You can ignore this part, this part is used for QE

File: ci-operator/step-registry/hypershift/mce/agent/metal3/conformance/hypershift-mce-agent-metal3-conformance-workflow.yaml (hunk 1 above)
Comment (jparrill) [reply ci-operator/step-registry/hypershift/mce/agent/metal3/conformance/hypershift-mce-agent-metal3-conformance-workflow.yaml:16]: ohh, gotcha, It's fine then

**Pull request merged**